### PR TITLE
Run GC multiple times if necessary

### PR DESCRIPTION
**Pull request by JoshCheek**
As suggested in https://github.com/rspec/rspec-core/pull/1950#issuecomment-99492550
Still leaves the conditional in, b/c Rbx didn't collect all the instances in my trial,
and JRuby wouldn't iterate over ObjectSpace to check whether the objects were collected.

### CODE REVIEW

**Comment by JoshCheek** [comment]
Hmm, should I put an ensure on that GC.enable? Frustratingly, it's a global var, so any change at all could break some other test.

I guess if any other tests wind up using that method, they'll pull it up to a more common location, and then could put a mutex on it or something. Since there's only one use case, that's maybe overkill for now.

File: spec/rspec/core/example_spec.rb [hunk 1]
```diff
@@ -344,11 +344,28 @@ def self.reliable_gc
...
 
+      def expect_gc(opts)
+        get_all = opts.fetch :get_all
+
+        GC.disable
+        opts.fetch(:event).call
+        expect(get_all.call).to eq(opts.fetch :pre_gc)
+        GC.enable
+
+        20.times do
```
Comment (myronmarston) [inline spec/rspec/core/example_spec.rb]: Do you mind adding a comment here like:

> # see discussion on https://github.com/rspec/rspec-core/pull/1950
> # for why it's necessary to do this multiple times

It's odd enough that I think it warrants this :).

**Comment by myronmarston** [comment]
> Hmm, should I put an ensure on that GC.enable? Frustratingly, it's a global var, so any change at all could break some other test.

Please either put it in `ensure` or add an `after` hook that does `GC.enable`.  `after` hooks are guaranteed to run so it would have the same effect (and `GC.enable` should be idempotent so multiple times shouldn't be a problem).

**Comment by JoshCheek** [comment]
Went with ensure, b/c I dislike that you would have to know that every time you call this method, you need to also add an after hook. With the ensure, it cleans up after itself.

WRT calling it multiple times, I just mean that the after hook could run when another thread expects it to be disabled. There aren't any right now, but if there are, then at least keeping all enabling and disabling in this method means we can put a mutex on it if we need.

**Comment by myronmarston** [comment]
:+1:  Thanks Josh!

**Pull request merged**